docs/source/satkill_explanation.rst

```diff
@@ -0,0 +1,240 @@
+        return labels, component_sizes
+
+
+    def generate_satellite_mask(im_deconv          : np.ndarray[float], 
+                                satellite_max_size : int, 
+                                e_cut              : float, 
+                                cut_type           : Optional[CutType]=CutType.abs) -> np.ndarray:
+        '''
+        An adaptation to the scikit-image (v0.24.0) function [1], identifies 
+        satellite energy depositions within deconvolution image by size
+        and proximity to other depositions.
+
+        The function takes a deconvolution z-slice, applies a mask based on the `e_cut` and `cut_type` to only
+        allow 0s and 1s for values passing the energy cut. 
+        It then uses the scipy ndimage library to identify different 'clusters' of 1s within the slice, and 
+        checks if these clusters are below the size considered for satellite deposits (`satellite_max_size`).
+        These satellite deposits are highlighted in a new mask, that is returned and used to remove them.
+
+        Returns the mask required to remove satellites as done in `richardson_lucy()`
+        
+        Parameters
+        ----------
+        im_deconv          : Deconvoluted 2D array
+        satellite_max_size : Maximum size of satellite deposit, above which they are considered 'real'.
+        e_cut              : Cut over the deconvolution output, arbitrary units or percentage
+        cut_type           : Cut mode to the deconvolution output (`abs` or `rel`) using e_cut
+                            `abs`: cut on the absolute value of the hits.
+                            `rel`: cut on the relative value (to the max) of the hits.
+
+        Returns
+        -------
+        array : Boolean mask of all labelled satellite deposits, True where satellites should be removed.
+
+        References
+        ----------
+        .. [1] https://github.com/scikit-image/scikit-image/blob/main/skimage/morphology/misc.py#L59-L151
+        '''
+        if cut_type is CutType.rel:
+            im_deconv = im_deconv / im_deconv.max()
+
+        # separate different regions below and above e_cut
+        # then label regions (components) appropriately and determine their sizes.
+        labels, component_sizes = collect_component_sizes(im_deconv >= e_cut)
+
+        # check if no satellites within deposit return False array
+        # (mask that removes no satellites).
+        if len(component_sizes) <= 2:
+            return np.full(im_deconv.shape, False)
+
+        # Find regions smaller than `satellite_max_size` and mask them, 
+        # ignoring the first region (background). Read gist for full explanation.
+        too_small      = component_sizes < satellite_max_size
+        too_small[0]   = False 
+        too_small_mask = too_small[labels]
+        return too_small_mask
+
+This mask is then used to remove satellites from 2D deconvolution arrays. Lets step through the logic!
```

Comment:
Add some minimal introduction with an overview of the method (1 or 2 phrases). Something like it uses these functions to produce a mask that is then used to suppress the satellites in the deconvolved image.